### ISSUE TITLE
login with Gmail failed with unclear message

### ISSUE THREAD

**Issue by humazed** [k9mail/k-9#2381]
### Actual behavior
When login with Gmail account I got this error message 
`username or password incorrect (command: *sensitive*; response: #3# [No,[ALERT],Please login via web browser: http://support.google.com/mail/accounts/answer/78754(Failure)])`
 which is not a very useful message.
after searching on google I found the solution is to https://www.google.com/settings/security/lesssecureapps
and enable access for less secure apps.

### Expected behavior
I should log into Gmail without much trouble, I know this might take a lot of time;
so I propose at least in the mean time to show the user more meaningful name with the link to this [page.](https://support.google.com/accounts/answer/6010255) as it's not convenient for first time user to search the web for a solution.

### Steps to reproduce
1. Login with Gmail account with [lesssecureapps](https://www.google.com/settings/security/lesssecureapps) turned off.

### Environment
K-9 Mail version: 5.205

Android version:6.0, 7.0, 7.1

Account type (IMAP, POP3, WebDAV/Exchange): Gmail

**Comment by rijuldhir** [comment]
Got this same message when I tried to Login with Gmail.
I would like to work on it.

**Comment by humazed** [comment]
I'm already working on it. and will have PR in one day; of course you're welcome to make your own PR.

**Comment by gloomy-ghost** [comment]
I am using two step verification and therefore can't modify the [lesssecureapps](https://www.google.com/settings/security/lesssecureapps) option. Maybe we should show [apppasswords](https://myaccount.google.com/apppasswords) instead, and consider oauth2 in the future.

**Comment by philipwhiuk** [comment]
Error response improvement is covered in my Enhanced Status Code work.

OAuth work is mostly implemented in terms of backend work. It needs some UI work to finish off, especially in terms of supporting both Google and none Google providers.

You should already be able to use app specific​ passwords.

**Comment by philipwhiuk** [comment]
Closing as I think existing issues cover all parts of this.

**Comment by humazed** [comment]
I was working on an enhanced dialog to show more clear error message and was getting ready to make PR should I proceed or do you think the existing solution is enough.

**Comment by philipwhiuk** [comment]
Feel free to still create a PR. There's issues #1058, #1538 and #2000 still open which your PR would work towards fixing.

**Comment by h-2-0** [comment]
I was getting the same error message and couldn't work out what was going on (I just came across this Github). At first I thought error message had to do with the 2 step verification process I had it turned it ON so I turned it OFF. But that didn't stop the error message from showing up again. 

After reading the contents of the link referred by the error message I turned OFF the Allow the Less Secure Apps to be able to open my mails. So I agree with @humazed, the error message should refer to the Google's Less Secure Apps page or directly to [https://myaccount.google.com/lesssecureapps](https://myaccount.google.com/lesssecureapps)
